table/tables/tables.go

```diff
@@ -237,36 +254,41 @@ func (t *Table) Truncate(ctx context.Context) error {
 }
 
 // UpdateRecord implements table.Table UpdateRecord interface.
-func (t *Table) UpdateRecord(ctx context.Context, h int64, currData []interface{}, newData []interface{}, touched []bool) error {
-	// if they are not set, and other data are changed, they will be updated by current timestamp too.
-	// set on update value
-	err := t.setOnUpdateData(ctx, touched, newData)
+func (t *Table) UpdateRecord(ctx context.Context, h int64, oldData []interface{}, newData []interface{}, touched map[int]bool) error {
```

Comment:
For update stmt, most of the columns may be not changed, so it is better to use a map instead a slice, also if process write only column, it need not to to make a slice copy.